src/Storages/StorageMerge.cpp

```diff
@@ -246,12 +249,63 @@ Pipe StorageMerge::read(
         if (query_info.query->as<ASTSelectQuery>()->sampleSize() && !storage->supportsSampling())
             throw Exception("Illegal SAMPLE: table doesn't support sampling", ErrorCodes::SAMPLING_NOT_SUPPORTED);
 
+        Aliases aliases;
         auto storage_metadata_snapshot = storage->getInMemoryMetadataPtr();
+        auto storage_columns = storage_metadata_snapshot->getColumns();
+
+        if (processed_stage == QueryProcessingStage::FetchColumns && !storage_columns.getAliases().empty())
+        {
+            NameSet required_columns_set;
+            std::function<void(ASTPtr)> extract_columns_from_alias_expression = [&](ASTPtr expr)
```

Comment:
Maybe we can reuse `ColumnAliasesVisitor`?

Reply:
Done.